### PR TITLE
Add Google Benchmark version 1.5.0

### PR DESCRIPTION
**Pull request by AndrewGaspar**
FYI @gartung

### CODE REVIEW

File: var/spack/repos/builtin/packages/benchmark/package.py [hunk 1]
```diff
@@ -30,7 +31,8 @@ class Benchmark(CMakePackage):
                     'MinSizeRel', 'Coverage'))
 
     depends_on("cmake@2.8.11:", type="build", when="@:1.1.0")
-    depends_on("cmake@2.8.12:", type="build", when="@1.2.0:")
+    depends_on("cmake@2.8.12:", type="build", when="@1.2.0:1.4")
```
Comment (gartung) [inline var/spack/repos/builtin/packages/benchmark/package.py]: The range should be @1.2.0:1.4.99 otherwise 1.4.1 will not match.

Comment (AndrewGaspar) [reply var/spack/repos/builtin/packages/benchmark/package.py]: Actually, "1.4" will match all versions of "1.4.x". "1.4.0" would limit it.

Comment (AndrewGaspar) [reply var/spack/repos/builtin/packages/benchmark/package.py]: This was new to me. I had made the same assumption.

Comment (gartung) [reply var/spack/repos/builtin/packages/benchmark/package.py]: OK. As long and the older version can be built I am fine with this change.

**Pull request merged**